substrate/frame/broker/src/coretime_interface.rs

```diff
@@ -146,3 +146,25 @@ impl CoretimeInterface for () {
 	#[cfg(feature = "runtime-benchmarks")]
 	fn ensure_notify_revenue_info(_when: RCBlockNumberOf<Self>, _revenue: Self::Balance) {}
 }
+
+/// Trait for getting the associated account and origin of a task.
+///
+/// For parachains, this is the sovereign account, which is controlled by the parachain
+/// itself.
+pub trait TaskAccountInterface {
+	/// The type for representing accounts.
+	type AccountId;
+
+	/// The overarching origin type.
+	type OuterOrigin: Into<Result<Self::TaskOrigin, Self::OuterOrigin>>;
+
+	/// The custom task origin. Given that all tasks on Polkadot are parachains this will most
+	/// likely be the `Parachain` origin.
+	type TaskOrigin;
+
+	/// Ensures that the origin is a task origin and returns the associated `TaskId`.
+	fn ensure_task_sovereign_account(o: Self::OuterOrigin) -> Result<TaskId, BadOrigin>;
+
+	/// Returns the associated account of a task.
+	fn sovereign_account(task: TaskId) -> Option<Self::AccountId>;
+}
```

Comment:
I just realized we don't need this at all. Sovereign accounts are represented as signed origin: https://github.com/paritytech/polkadot-sdk/blob/d539778c3cc4e0376769472fdad37856f4051dc5/polkadot/xcm/xcm-builder/src/origin_conversion.rs#L27-L29

This will simplify a lot of the code.

Reply:
OK cool, I'll abandon my review for now then - I'm not all the way through but just have some comments